### PR TITLE
DEV-4998: chess readme and audit

### PR DESCRIPTION
**Pull request by MSilva95**
chess fix audit and readme

### CODE REVIEW

File: subjects/mobile-dev/chess/README.md [hunk 1]
```diff
@@ -12,12 +10,40 @@ So whether you're a seasoned grandmaster or a newcomer to the game, join the ran
...
 
-1. Research different chess rules and app designs to get ideas for your own app.
-2. Think about how you will handle game state and moves within the app, and how you will ensure that the game is played fairly and accurately.
-3. Implement backend according to the swagger:
+- **Restart**, to allow players to restart the game
 
-By taking these factors into account, you can create a successful and engaging chess app that meets the objectives of the exercise.
+- **Board Reseet**, to enable players to reset the board without restarting the entire game
+- **Story Mode**: Design a single-player story mode where players can engage in chess matches against AI opponents, progressing through a series of challenges or levels.
+- **Bots**: Include AI-powered bots that players can play against in offline mode or when waiting for online opponents.
```
Comment (nprimo) [inline subjects/mobile-dev/chess/README.md:48]: This bonus point looks the same as **Versus Computer Mode**: what is the difference? Should we make them the same?

File: subjects/mobile-dev/chess/audit/README.md [hunk 2]
```diff
@@ -1,19 +1,51 @@
 #### Functional
 
-#### In order to run and hot reload app either on emulator or device, follow the [instructions](https://docs.flutter.dev/get-started/test-drive?tab=androidstudio#run-the-app).
+#### In order to run and hot reload the app either on emulator or device, follow the [instructions](https://docs.flutter.dev/get-started/test-drive?tab=androidstudio#run-the-app).
```
Comment (nprimo) [inline subjects/mobile-dev/chess/audit/README.md:3]: This paragraph is a `>` section in the other audits

File: subjects/mobile-dev/chess/README.md [hunk 3]
```diff
@@ -12,12 +10,40 @@ So whether you're a seasoned grandmaster or a newcomer to the game, join the ran
...
+Your task is to develop a mobile app that allows users to play chess with each other. This is a fullstack app so you will need to implement both backend and frontend. You may use any desired backend technology as long as you're following the [backend routes.](https://github.com/alem-01/chess).
+
+Make sure that your app has the following requirements:
+
+- Must have UI screens, including at least the following:
+  - A Main menu
+  - A Game board
+  - A waiting room
+- Must implement the logic to validate moves and enforce the rules of chess, disallowing illegal moves.
+- Must Ensure that white pieces always play first in each game.
```
Comment (nprimo) [inline subjects/mobile-dev/chess/README.md:24]: The word `Ensure` should be lowercase

File: subjects/mobile-dev/chess/audit/README.md [hunk 4]
```diff
@@ -1,19 +1,51 @@
...
+##### Try performing illegal moves (e.g., moving a pawn 3 squares up, castling after moving the king, moving a rook diagonally).
+
+###### Does the app prevent illegal moves?
 
 ###### After a move is played, does the other player receive the move?
 
-###### Play the game until one player is mated, does the game stop?
+##### Play the game until one player is mated.
+
+###### Does the game ends and the player who mated the other win?
```
Comment (nprimo) [inline subjects/mobile-dev/chess/audit/README.md:33]: `ends` should be `end`

File: subjects/mobile-dev/chess/audit/README.md [hunk 5]
```diff
@@ -1,19 +1,51 @@
...
-###### Check the backend. When player is the waiting room and someone joins, does the game create UUID for the game session?
+- A Main menu
+- A Game board
+- A waiting room
 
-###### When 2 players are join game session, can a player with white pieces start the game?
+###### Does the app have all the UI screens requested?
 
-###### Try doing illegal moves, i.e. moving pawn 3 squares up, castling after moving the king, moving rook diagonally, etc. Does the app prevent such illegal moves?
+###### Does the app have a multiplayer mode for real-time matches which enables players to challenge and play against other?
```
Comment (nprimo) [inline subjects/mobile-dev/chess/audit/README.md:15]: I would rephrase with the following:
```
... and play against each other?
```

File: subjects/mobile-dev/chess/README.md [hunk 6]
```diff
@@ -12,12 +10,40 @@ So whether you're a seasoned grandmaster or a newcomer to the game, join the ran
...
-Here are some additional considerations to keep in mind as you work on this exercise:
+- **Player X Won**, after determining the winner of the game, send a message to both players indicating which player has won.
 
-1. Research different chess rules and app designs to get ideas for your own app.
-2. Think about how you will handle game state and moves within the app, and how you will ensure that the game is played fairly and accurately.
-3. Implement backend according to the swagger:
+- **Restart**, to allow players to restart the game
 
-By taking these factors into account, you can create a successful and engaging chess app that meets the objectives of the exercise.
+- **Board Reseet**, to enable players to reset the board without restarting the entire game
```
Comment (nprimo) [inline subjects/mobile-dev/chess/README.md:46]: There is a small typo: `Reseet` -> `Reset`

Comment (nprimo) [reply subjects/mobile-dev/chess/README.md:46]: From a game perspective, this bonus point is not clear to me: reset the board without starting the game - the pieces go back to the starting positions and what about the `without restarting the entire game`, is that referring to the app?

File: subjects/mobile-dev/chess/audit/README.md [hunk 7]
```diff
@@ -1,19 +1,51 @@
...
+###### +Does the app allow players to restart the game?
+
+###### +Can players reset the board without restarting the entire game?
+
+###### +Is there a single-player story mode where players can engage in chess matches against AI opponents, progressing through a series of challenges or levels?
+
+###### +Are there AI-powered bots that players can play against in offline mode or when waiting for online opponents?
 
-###### Do chess moves come with appropriate Long algebraic notation?
+###### +Does the app have a mode where players can play against a computer AI with adjustable difficulty levels?
```
Comment (nprimo) [inline subjects/mobile-dev/chess/audit/README.md:51]: I feel there is no difference between this question and the one above

**Review by nprimo — CHANGES_REQUESTED** [review]
I left some comments inline

File: subjects/mobile-dev/chess/README.md (hunk 1 above)
Comment (MSilva95) [reply subjects/mobile-dev/chess/README.md:48]: done

File: subjects/mobile-dev/chess/audit/README.md (hunk 2 above)
Comment (MSilva95) [reply subjects/mobile-dev/chess/audit/README.md:3]: done

File: subjects/mobile-dev/chess/audit/README.md [hunk 8]
```diff
@@ -1,19 +1,51 @@
...
+
+###### Does the game ends and the player who mated the other win?
+
+###### Does the backend generate a unique identifier (UUID) for each game session when a player joins the waiting room, ensuring proper identification and management of individual game sessions?
+
+#### Bonus
+
+###### +When the game reaches a terminal state (checkmate, stalemate, or draw), does the app send a message to both players indicating that the game is over?
+
+###### +After determining the winner of the game, does the app send a message to both players indicating which player has won?
```
[the remark this answers is not in the document]
Comment (MSilva95) [reply subjects/mobile-dev/chess/audit/README.md:41]: The first one is to make sure that the game actually ends at some point.
The second one involves sending a message which is treated as a bonus in the Readme

File: subjects/mobile-dev/chess/README.md (hunk 1 above)
Comment (MSilva95) [reply subjects/mobile-dev/chess/README.md:44]: done

File: subjects/mobile-dev/chess/audit/README.md (hunk 2 above)
Comment (MSilva95) [reply subjects/mobile-dev/chess/audit/README.md]: done

File: subjects/mobile-dev/chess/README.md (hunk 3 above)
Comment (MSilva95) [reply subjects/mobile-dev/chess/README.md]: done

File: subjects/mobile-dev/chess/audit/README.md (hunk 4 above)
Comment (MSilva95) [reply subjects/mobile-dev/chess/audit/README.md]: done

File: subjects/mobile-dev/chess/audit/README.md (hunk 5 above)
Comment (MSilva95) [reply subjects/mobile-dev/chess/audit/README.md]: done

File: subjects/mobile-dev/chess/README.md (hunk 6 above)
Comment (MSilva95) [reply subjects/mobile-dev/chess/README.md]: Actually yes it makes sense, it is the same as a restart. i removed it here and in the audit

File: subjects/mobile-dev/chess/audit/README.md (hunk 7 above)
Comment (MSilva95) [reply subjects/mobile-dev/chess/audit/README.md]: I left only the Bots bonus as it seems a better option to leave other than the"vs Computer" removed it from the readme as well

File: subjects/mobile-dev/chess/audit/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (nprimo) [reply subjects/mobile-dev/chess/audit/README.md:41]: In both questions there is a `does the app send a message` - the first one should be more vague I guess, but at the same time, it should be something already included in the mandatory part (I guess)

File: subjects/mobile-dev/chess/README.md [hunk 9]
```diff
@@ -12,12 +10,35 @@ So whether you're a seasoned grandmaster or a newcomer to the game, join the ran
...
+- Research different chess rules and app designs to get ideas for your own app.
+- Think about how you will handle game state and moves within the app, and how you will ensure that the game is played fairly and accurately.
 
-1. Research different chess rules and app designs to get ideas for your own app.
-2. Think about how you will handle game state and moves within the app, and how you will ensure that the game is played fairly and accurately.
-3. Implement backend according to the swagger:
+### Bonus
 
-By taking these factors into account, you can create a successful and engaging chess app that meets the objectives of the exercise.
+- **Game Over**, when the game reaches a terminal state (checkmate, stalemate, or draw), we can send a message to both players indicating that the game is over.
```
Comment (nprimo) [inline subjects/mobile-dev/chess/README.md:40]: This point and the below ones seems exactly the same ask to me (same thing as the open comment in the audit file)

File: subjects/mobile-dev/chess/audit/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (MSilva95) [reply subjects/mobile-dev/chess/audit/README.md]: i removed the repetition and made this part mandatory as it makes more sense to have it done in the mandatory part

File: subjects/mobile-dev/chess/resources/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+WORKDIR /app
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+COPY . .
+RUN go build -o /app/main .
+
+
+FROM ubuntu:20.04
```
Comment (nprimo) [inline subjects/mobile-dev/chess/resources/Dockerfile:11]: I have commented from line 11 to line 14 and I still get the same result. 

I don't understand the value of having this additional lines in the Dockerfile

File: subjects/mobile-dev/chess/resources/README.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+- [References](#references)
+
+## Local setup
+
+There are two major ways to locally setup the project to have it up and running:
+
+- [Build from source](#build-from-source)
+- [Build with Docker](#build-from-docker)
+
+### Build from source
```
Comment (nprimo) [inline subjects/mobile-dev/chess/resources/README.md:19]: Running the instructions below, I don't get the same result (the UUID) that I get with the Docker section.

```
╰─± go run .
2023/06/05 16:05:57 running chess server on port :8080...

```

File: subjects/mobile-dev/chess/resources/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,15 @@
+FROM golang:1.19.1-buster AS builder
+
+WORKDIR /app
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+COPY . .
```
Comment (nprimo) [inline subjects/mobile-dev/chess/resources/Dockerfile:7]: This line also might be removed - it should not effect the final result

File: subjects/mobile-dev/chess/resources/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,15 @@
+FROM golang:1.19.1-buster AS builder
```
Comment (nprimo) [inline subjects/mobile-dev/chess/resources/Dockerfile:1]: this might be simplified to `FROM golang:1.19.1-buster`

File: subjects/mobile-dev/chess/resources/Dockerfile (hunk 10 above)
Comment (MSilva95) [reply subjects/mobile-dev/chess/resources/Dockerfile]: will close this until we work on it!

File: subjects/mobile-dev/chess/resources/Dockerfile (hunk 13 above)
Comment (MSilva95) [reply subjects/mobile-dev/chess/resources/Dockerfile]: will close this until we work on it!

**Review by nprimo — APPROVED** [review]
LGTM
